### PR TITLE
[Fix]: TSConverter handles call ops with multiple outputs

### PR DESCRIPTION
**Pull request by jiashenC**
#### Issue
* Current call ops does not handle IR with multiple outputs. If an op has multiple outputs, we add an implicit unpack to map output. E.g.,
```
%5 : Tensor, %6 : Tensor = aten::max(%x.1, %3, %4), scope: export.test_converter.M:: # /data/users/jiashenc/pytorch/test/export/test_converter.py:774:20
```
* There are some cases that `prim::If` sub-blocks do not return any outputs. E.g., 
```
%9 : bool = aten::gt(%8, %3), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # <string>:5:9
   = prim::If(%9), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # <string>:5:2
    block0():
      -> ()
    block1():
       = prim::RaiseException(%5, %4), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # <string>:5:2
      -> ()
```

#### Test Plan
We did an exhaustive search of all torch APIs that can return multiple outputs. We sample some of common ones and add new test cases based on those.
* `pytest test/export/test_converter.py -s -k test_ts2ep_multi_outputs_on_call_ops`

#### Appendix
* aten ops that return multiple outputs.
```
aten._batch_norm_impl_index
aten._batch_norm_no_update
aten._batch_norm_with_update
aten._batch_norm_with_update_functional
aten._cudnn_rnn
aten._efficient_attention_backward
aten._efficient_attention_forward
aten._embedding_bag
aten._embedding_bag_forward_only
aten._flash_attention_backward
aten._flash_attention_forward
aten._fused_adam
aten._fused_dropout
aten._fused_moving_avg_obs_fq_helper
aten._linalg_det
aten._linalg_eigh
aten._linalg_slogdet
aten._linalg_solve_ex
aten._linalg_svd
aten._native_batch_norm_legit
aten._native_batch_norm_legit_functional
aten._native_batch_norm_legit_no_training
aten._pack_padded_sequence
aten._prelu_kernel_backward
aten._scaled_dot_product_efficient_attention
aten._scaled_dot_product_efficient_attention_backward
aten._scaled_dot_product_flash_attention
aten._scaled_dot_product_flash_attention_backward
aten._scaled_dot_product_flash_attention_for_cpu
aten._scaled_dot_product_flash_attention_for_cpu_backward
aten._thnn_fused_lstm_cell
aten._thnn_fused_lstm_cell_backward_impl
aten._unique2
aten._weight_norm_interface
aten.adaptive_max_pool2d
aten.adaptive_max_pool3d
aten.aminmax
aten.batch_norm_backward
aten.convolution_backward
aten.cudnn_batch_norm
aten.cudnn_batch_norm_backward
aten.cummax
aten.cummin
aten.fractional_max_pool2d
aten.frexp
aten.grid_sampler_2d_backward
aten.grid_sampler_3d_backward
aten.gru
aten.linalg_cholesky_ex
aten.linalg_eig
aten.linalg_inv_ex
aten.linalg_ldl_factor_ex
aten.linalg_lu
aten.linalg_lu_factor_ex
aten.linalg_qr
aten.linear_backward
aten.log_sigmoid_forward
aten.lstm
aten.lu_unpack
aten.max
aten.max_pool2d_with_indices
aten.max_pool3d_with_indices
aten.median
aten.min
aten.miopen_batch_norm
aten.miopen_batch_norm_backward
aten.mkldnn_rnn_layer
aten.mkldnn_rnn_layer_backward
aten.mode
aten.multilabel_margin_loss_forward
aten.nanmedian
aten.native_batch_norm
aten.native_batch_norm_backward
aten.native_dropout
aten.native_group_norm
aten.native_group_norm_backward
aten.native_layer_norm
aten.native_layer_norm_backward
aten.nll_loss2d_forward
aten.nll_loss_forward
aten.quantized_gru
aten.quantized_lstm
aten.rnn_relu
aten.rnn_tanh
aten.sort
aten.std_mean
aten.topk
aten.triangular_solve
aten.unique_dim
aten.var_mean
```

### CODE REVIEW

File: torch/_export/converter.py [hunk 1]
```diff
@@ -620,11 +628,10 @@ def convert_prim_If(self, node: torch._C.Node):
 
         cond_node = self.fx_graph.call_function(torch.cond, args, {})
 
-        output_name = node.output().debugName()
-        self.name_to_node[output_name] = cond_node
-
-    def convert_aten_Bool(self, node: torch._C.Node):
-        self._convert_as_noop(node)
+        # prim::If can also have zero output.
```
Comment (jiashenC) [inline torch/_export/converter.py:631]: @BoyuanFeng Probably need to incorporate your changes here.

File: torch/_export/converter.py [hunk 2]
```diff
@@ -678,6 +685,9 @@ def convert_node(self, node: torch._C.Node):
         handler_func = getattr(self, handler_func_name, self.convert_call_function_op)
         handler_func(node)
 
+    def convert_prim_RaiseException(self, node: torch._C.Node):
```
Comment (jiashenC) [inline torch/_export/converter.py:688]: @BoyuanFeng Here as well.

File: torch/_export/converter.py [hunk 3]
```diff
@@ -620,11 +628,10 @@ def convert_prim_If(self, node: torch._C.Node):
...
 
-        output_name = node.output().debugName()
-        self.name_to_node[output_name] = cond_node
-
-    def convert_aten_Bool(self, node: torch._C.Node):
-        self._convert_as_noop(node)
+        # prim::If can also have zero output.
+        if node.outputsSize() == 1:
+            output_name = node.output().debugName()
+            self.name_to_node[output_name] = cond_node
```
Comment (angelayi) [inline torch/_export/converter.py:634]: what about the multiple output case?

File: torch/_export/converter.py [hunk 4]
```diff
@@ -705,9 +715,13 @@ def convert_graph_outputs(self):
...
-        self.fx_graph.output(
-            args[0]
-        )  # Get rid of an extra list wrapped around final output.
+        if len(args) == 1:
+            self.fx_graph.output(
+                args[0]
+            )  # Get rid of an extra list wrapped around final output.
+        else:
+            # Sub-block of prim::If can have zero output.
+            self.fx_graph.output([])
```
Comment (angelayi) [inline torch/_export/converter.py:724]: can you send an example for this?

Comment (jiashenC) [reply torch/_export/converter.py:724]:
```
graph(%x : Float(4, 4, strides=[4, 1], requires_grad=0, device=cpu),
      %y.1 : Float(1, 1, 10, 10, strides=[100, 100, 10, 1], requires_grad=0, device=cpu)):
  %2 : __torch__.export.test_converter.___torch_mangle_9.M = prim::CreateObject()
  %3 : int = prim::Constant[value=2]()
  %4 : NoneType = prim::Constant(), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool
  %5 : str = prim::Constant[value="AssertionError: "](), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool
  %6 : int[] = prim::Constant[value=[2, 2]]()
  %7 : int[] = aten::size(%y.1), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # /data/users/jiashenc/pytorch/torch/nn/functional.py:1266:50
  %8 : int = aten::len(%7), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # <string>:5:9
  %9 : bool = aten::gt(%8, %3), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # <string>:5:9
   = prim::If(%9), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # <string>:5:2
    block0():
      -> ()
    block1():
       = prim::RaiseException(%5, %4), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # <string>:5:2
      -> ()
  %10 : Tensor, %11 : Tensor = aten::adaptive_max_pool2d(%y.1, %6), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool # /data/users/jiashenc/pytorch/torch/nn/functional.py:1267:11
  %12 : (Tensor, Tensor) = prim::TupleConstruct(%10, %11), scope: export.test_converter.M::/torch.nn.modules.pooling.AdaptiveMaxPool2d::pool
  %13 : ((Tensor, Tensor)) = prim::TupleConstruct(%12), scope: export.test_converter.M::
  return (%13)
```

**Review by angelayi — APPROVED** [review]

**Pull request closed without merging**